### ISSUE TITLE
`am` records its draws over two different windows: `samples.txt` starts at `burn_in`, the trajectories and `combined_params.txt` at `burn_in + adaptive`

### ISSUE THREAD

**Issue by wshlavacek** [lanl/PyBNF#772]
## What happens

`fit_type = am` records its draws over **two different windows**, depending on which output you
read:

| output | window it starts at |
| --- | --- |
| `Results/samples.txt` (`sample_pset`, `adaptive_mcmc.py:337`) | `burn_in` |
| the trajectory buffers (`adaptive_mcmc.py:313`) | `valid_range` = `burn_in + adaptive` |
| `Results/A_MCMC/Runs/combined_params.txt` (`combine_chains_params`, the `[self.adaptive:]` slice) | `valid_range` |
| the ArviZ bridge (documented in `config_keys.rst` as reading `params_*.txt` for `am`) | `valid_range` |

So the samples file carries the `adaptive` window and the other three do not. Measured on a run
with `burn_in = 60`, `adaptive = 60`, `sample_every = 2`, `max_iterations = 300`, 3 chains:

```
Results/samples.txt rows  : 360
combined_params.txt rows  : 270
-> samples.txt carries 90 adaptation-phase draws (25% of it)
```

Since #771 the histograms and credible intervals are built from `samples.txt`, so they summarize
the wider window while a user's trajectories and their `InferenceData` summarize the narrower one.
Two summaries of "the posterior" from the same run are computed from different draw sets, and
nothing says so.

## Why this is not simply a bug

The wider window is defensible. During the `adaptive` iterations the chain is running a
fixed-step Metropolis kernel against the same posterior — `_update_covariance`-style adaptation
only takes over at `iteration >= burn_in + adaptive` (`adaptive_mcmc.py:662`) — so those draws
come from a valid MH kernel, just one that had not adapted yet. Discarding them is a choice about
efficiency and burn-in convention, not about validity.

This was decided deliberately when #771 landed: the credible intervals summarize the post-`burn_in`
window, matching what `samples.txt` means for every other sampler, and `samples.txt` was left
alone. This issue is to record the inconsistency rather than to assert which window is right.

## What to decide

1. Whether `am`'s outputs should agree on one window, and which. Moving `sample_pset` to
   `valid_range` is a one-line change but alters `samples.txt` for every `am` user, and it breaks
   `test_am_samples_a_flat_posterior_evenly_up_to_the_walls`, which deliberately sets
   `adaptive = iterations - 3` to hold the fixed-step branch for a whole run (#709) and would then
   collect almost no samples. That test would need reworking as part of any such change.
2. Failing that, whether the difference should simply be documented, in `config_keys.rst` next to
   the existing note about the ArviZ bridge reading `params_*.txt` for `am`.

## Where

`pybnf/algorithms/samplers/adaptive_mcmc.py`: the `sample_pset` gate at line 337 (post-`burn_in`)
against the trajectory gate at line 313 and the `[self.adaptive:]` slice in
`combine_chains_params`. There is a comment at the `sample_pset` gate pointing here.

Found while fixing #771.

**Comment by wshlavacek** [comment]
Checked the paper, and it settles more of this than expected.

**Source.** Neumann, Lin, Mallela, Miller, Colvin, Duprat, Chen, Hlavacek, Posner, *Implementation of a
practical Markov chain Monte Carlo sampling algorithm in PyBioNetFit*, Bioinformatics 2022, 38(6),
1770–1772, [doi:10.1093/bioinformatics/btac004](https://doi.org/10.1093/bioinformatics/btac004). It is
cited in `CITATION.cff` but not in the Adaptive MCMC section of `docs/algorithms.rst`, which points only
at Andrieu and Thoms (2008). The phases are indeed in the supplementary text, not the three-page note —
free full text including the supplement is at [arXiv:2109.14445](https://arxiv.org/abs/2109.14445), in
the section immediately before "Assessing the efficiency of MCMC sampling".

**The paper defines four phases, from three settings** (`n_burn-in`, `n_adaptive`, `n_stationary`):

1. **Burn-in**, to `n_burn-in` — proposals from the user's fixed Σ₀; the goal is to relax into a
   high-density region. Can be skipped by setting it to 0.
2. **Mixing**, to `n_burn-in + n_adaptive` — sampling continues on Σ₀. The paper is explicit that the
   `n_adaptive` samples gathered here are what the empirical covariance Σ̂ is computed *from*; it asks the
   user to check that the acceptance rate over this window was high enough to learn a covariance at all.
3. **Adaptation**, from `n_burn-in + n_adaptive` — Σ̂ initializes Σ*, then Σ* and the prefactor f are
   updated on the fly under a 1/n decaying weight, targeting an acceptance rate of 0.234.
4. **Stationary**, from `n_stationary` — where, in the paper's words, "samples will begin to be collected
   for the purpose of defining the posterior". It recommends `n_stationary − (n_burn-in + n_adaptive)` be
   at least 10³, and reports using > 10⁴ in the paper's own test applications.

**Against the implementation:**

| paper | boundary | PyBNF |
| --- | --- | --- |
| burn-in ends | `n_burn-in` | `burn_in` |
| mixing ends / adaptation begins | `n_burn-in + n_adaptive` | `valid_range` (`adaptive_mcmc.py:52`; the on-the-fly update starts at `:669`) |
| posterior collection begins | `n_stationary` | **no counterpart** |

So `am` implements three of the four boundaries. There is no `n_stationary` key, and nothing in the
config surface or the docs corresponds to it — `docs/config_keys.rst` describes `adaptive` only as
iterations spent "collecting data ... for calcualtion of the differential matrix", which is the paper's
mixing phase, with no mention of a later point at which posterior collection starts.

That means **both** windows in the table at the top of this issue begin earlier than the paper's
posterior-collection point, not just the wider one:

- `samples.txt` starts at `burn_in`, so it includes the mixing phase — the samples the paper describes as
  the training data for Σ̂.
- the trajectories and `combined_params.txt` start at `burn_in + adaptive`, which is where the paper's
  *adaptation* phase begins, still at least 10³ iterations (the paper's own recommendation) before the
  stationary phase.

**Bearing on #771/#773.** The credible intervals shipped there are built from `samples.txt`, i.e. the
widest of the three windows. The reasoning recorded on that PR — that the adaptive window runs a valid
fixed-step Metropolis kernel, so its draws are legitimate — is sound as general MCMC theory, and the
paper does not contradict it as a statement about validity. But the paper does prescribe something
stricter for this sampler specifically, and it is the sampler's own specification. Worth deciding whether
the shipped intervals should follow it.

**Options this opens, in increasing order of size:**

1. Document the phases. Fold the four-phase description and the `am` paper citation into
   `docs/algorithms.rst` and the `burn_in` / `adaptive` entries in `docs/config_keys.rst`, and say plainly
   which window each output file covers. This is worth doing whichever of the below is chosen.
2. Align the existing outputs on one window, as this issue originally proposed — noting that neither
   current choice is the paper's.
3. Add the missing `n_stationary` setting, so the sampler exposes the boundary the paper defines and
   posterior collection starts there. This is the only option that makes PyBNF's `am` match its own
   published specification, and it is also the largest: a new config key, a changed default for what
   `samples.txt` contains, and rework of the flat-box test from #709, which sets `adaptive = iterations - 3`
   on purpose.

No opinion pushed on which; flagging that the decision is now better informed than when this was filed.

**Comment by wshlavacek** [comment]
## Notes toward the careful analysis

Recording the reasoning so it does not have to be re-derived. This is a proposal for discussion,
not a decision — the empirical question at the bottom is the one that should actually settle it.

### The part that is not a judgment call

Whatever window is chosen, every output should use **one** definition of "the posterior sample" for a
run. Today `samples.txt`, the trajectory buffers, `combined_params.txt`, the ArviZ object, and (since
#773) the histograms and credible intervals do not all cover the same draws. A user comparing a
credible interval against a trace plot is comparing different data. That is wrong independently of
which window is right, and it is the part worth fixing regardless of how the rest lands.

### Statistical status of each phase

The four phases are not simply "discard" and "keep" — they differ in *why* their draws are or are not
trustworthy.

**Burn-in.** Discard. Uncontroversial.

**Mixing (fixed Σ₀).** Valid but poor. A fixed-kernel Metropolis chain is a genuine Markov chain with
the posterior as its invariant distribution, so conditional on the chain having actually relaxed,
these are legitimate posterior draws. There is also no circularity in reusing them: the draws do not
depend on Σ̂, so using them to build Σ̂ for *later* iterations does not taint them. The objection is
practical — Σ₀ is the user's untuned guess, so these are the most autocorrelated draws in the run and
the most likely to still be transient. Lowest effective sample size per draw, highest bias risk.

**Adaptation.** Asymptotically valid, and this is the load-bearing point. An adaptive chain is not a
Markov chain (the proposal depends on the whole history), so validity does not come from standard MCMC
theory; it comes from the adaptive-MCMC ergodicity results — Haario et al. (2001), Roberts and
Rosenthal (*Coupling and ergodicity of adaptive MCMC algorithms*, J. Appl. Probab. 44(2):458–475,
2007) — which require **diminishing adaptation** and **containment**.

Checked the first against this implementation and it holds: the updates at `adaptive_mcmc.py:688-691`
are Robbins–Monro with 1/n gains, and `stablizingCov` (`:668`) keeps Σ positive definite. So the
adaptation-phase draws are covered by the theory. The adaptive-MCMC literature exists precisely so
that this phase does not have to be thrown away.

One nuance cuts the other way, and it is worth knowing before choosing a default. The gain on Σ is
`1/(1 + iteration − burn_in)`, already ≈ 1/`adaptive` at the seeding iteration, so Σ moves slowly from
its seed. But the gain on the scale prefactor `f` is `1/(1 + iteration − adaptive − burn_in)`, which
starts at **1** and decays — `f` is seeded at 2.38²/N (`:681`) and then moves fast. So the overall
proposal scale is changing rapidly in the first stretch after adaptation begins. That stretch is the
least trustworthy part of the run, and it is a concrete mechanism behind the paper's Δn ≥ 10³
recommendation rather than just a rule of thumb.

**Stationary.** The clean draws.

### What this implies

1. **Validity does not require the paper's discard.** Both windows currently in use produce
   asymptotically valid draws. This is a quality-and-consistency problem, not "PyBNF computes wrong
   posteriors". Worth saying plainly so the fix is not over-scoped.
2. **Quality argues for collecting after the scale has settled**, which is what `n_stationary` encodes.
3. **The boundary should be explicit and user-set.** The paper carries three settings because the right
   answer is problem-dependent; PyBNF derives its window instead, and gives the user no way to say
   "start collecting here."
4. **But a fixed 10³/10⁴ is a heuristic standing in for "has it converged?"** — and PyBNF can answer
   that question directly. It already computes rank-normalized split-R̂ and bulk/tail ESS (Vehtari et
   al. 2021). The strongest behavior is to collect from an explicit `n_stationary` **and report the
   diagnostics computed on exactly the retained draws**, so a user learns whether their choice was
   good instead of receiving silent intervals.

Proposed shape, for discussion: an explicit `n_stationary` key (defaulting to today's `valid_range`
for compatibility, or to the paper's recommendation if a default change is acceptable), every output
reading it, and R̂/ESS reported on the retained set.

**Interim ranking**, if the full change is not taken soon: the post-`burn_in` window is the weakest of
the three, because it is the only one that includes the untuned-Σ₀ stretch. Moving the summaries to
`valid_range` would at least align them with the trajectories and the ArviZ object and drop the worst
draws, without adding any config surface.

### Open questions for the analysis

- **Containment.** Diminishing adaptation is verified above; containment is not. A bounded
  `uniform_var` box makes it plausible; unbounded `normal_var` priors are less obvious.
- **Count or diagnostic?** Whether `n_stationary` is best expressed as an iteration count or as a
  convergence trigger on the existing R̂/ESS machinery.
- **The empirical question, which is cheap and should probably come first:** how much does including
  the mixing and adaptation phases actually move a credible interval? This is directly measurable
  against the two oracles already in the test harness — the flat box over `uniform_var [0, 1]`, whose
  posterior is exactly Uniform(0, 1), and `make_linear_gaussian_config`, whose posterior is Gaussian in
  closed form. If the answer is "by less than the Monte Carlo error", the consistency fix alone is
  enough and the rest is bookkeeping. If it is not, that measurement makes the case for `n_stationary`
  far better than any argument from theory.
